openff/bespokefit/cli/cache.py

```diff
@@ -0,0 +1,195 @@
+import hashlib
+import json
+import uuid
+from typing import Optional, Union
+
+import click
+import redis
+import rich
+from click_option_group import optgroup
+from openff.qcsubmit.results import (
+    OptimizationResultCollection,
+    TorsionDriveResultCollection,
+)
+from rich import pretty
+from rich.padding import Padding
+from rich.progress import track
+
+from openff.bespokefit.cli.utilities import create_command, print_header
+from openff.bespokefit.executor.services import settings
+from openff.bespokefit.executor.services.qcgenerator.cache import _canonicalize_task
+from openff.bespokefit.executor.utilities.redis import is_redis_available, launch_redis
+from openff.bespokefit.schema.data import LocalQCData
+from openff.bespokefit.schema.tasks import task_from_result
+
+
+@click.group("cache")
+def cache_cli():
+    # TODO do we want the redis database to be saved in a standard location as it is directory dependent?
+    """Commands to manually update the qc data cache."""
+
+
+def update_from_qcsubmit_options(
+    launch_redis_if_unavailable: Optional[bool] = True,
+):
+    return [
+        optgroup("Input Configuration"),
+        optgroup.option(
+            "--input",
+            "input_file_path",
+            type=click.Path(exists=True, file_okay=True, dir_okay=False),
+            help="The serialised openff-qcsubmit file.",
+            required=True,
+        ),
+        optgroup.group("Storage configuration"),
+        optgroup.option(
+            "--launch-redis/--no-launch-redis",
+            "launch_redis_if_unavailable",
+            help="Whether to launch a redis server if an already running one cannot be "
+            "found.",
+            required=launch_redis_if_unavailable is None,
+            default=launch_redis_if_unavailable,
+            show_default=launch_redis_if_unavailable is not None,
+        ),
+    ]
+
+
+def _update(input_file_path: str, launch_redis_if_unavailable: bool):
+    """
+    The main worker function which updates the redis cache with qcsubmit results objects.
+    """
+
+    pretty.install()
+    console = rich.get_console()
+    print_header(console)
+
+    console.print(Padding("1. gathering QCSubmit results", (0, 0, 1, 0)))
+
+    with console.status("loading results file"):
+        try:
+            qcsubmit_result = TorsionDriveResultCollection.parse_file(input_file_path)
```

Comment:
@SimonBoothroyd I think this could be simplified if we a type field to the collection objects in qcsubmit.